Validating JSON Properties File Reads Prior to Processing

mirichmo opened the pull request:
Fixes #2342 and #2346 
1. I added an additional sanity check to the JSON file processing during reads to better handle empty files such as "{}".
2. I Refactored the ExecutionPolicy tests to centralize them in the new location, remove duplication, and handle windows vs. non-windows execution using the new try-catch technique.

mirichmo committed:
Fixing PowerShellProperties.json reads to do a basic validity check prior to processing the JSON

mirichmo committed:
Moving ExecutionPolicy tests to Microsoft.PowerShell.Security and refactoring test skipping for Windows.

msftclas commented:
Hi __@mirichmo__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Mike Richmond). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

mirichmo commented:
@vors Please review

vors commented:
I'm curious, why the whole
files changed? It could be due to tab vs spaces or line ends. Can you check?

mirichmo commented:
I'm not quite sure what you mean. I moved a bunch of ExecutionPolicy tests from engine\ExecutionPolicy.Tests.ps1 to modules\Microsoft.powershell.security\ExecutionPolicy.Tests.ps1. I also removed duplicate tests from engine\ExecutionPolicy.Tests.ps1.
